### PR TITLE
Move application handler call out of synchronized block.

### PR DESCRIPTION
**Pull request by boaks**
Though handler may call other synchronized stuff, the call is moved out
of the synchronized block.

Signed-off-by: Achim Kraus <achim.kraus@bosch-si.com>

### CODE REVIEW

File: scandium-core/src/main/java/org/eclipse/californium/scandium/DTLSConnector.java [hunk 1]
```diff
@@ -716,8 +718,10 @@ private void processApplicationDataRecord(final Record record) {
 						// thus, the handshake seems to have been completed successfully
 						connection.handshakeCompleted(record.getPeerAddress());
 						session.markRecordAsRead(record.getEpoch(), record.getSequenceNumber());
-						// finally, forward de-crypted message to application layer
-						handleApplicationMessage(message, session);
+						// create application message, if handler is set.
+						if (messageHandler != null) {
```
Comment (sbernard31) [inline scandium-core/src/main/java/org/eclipse/californium/scandium/DTLSConnector.java]: Should we keep this test ?

Comment (boaks) [reply scandium-core/src/main/java/org/eclipse/californium/scandium/DTLSConnector.java]: If agreed, that such a connector without messageHandler doesn't make sense, we can remove it and just keep the check below.

Comment (sbernard31) [reply scandium-core/src/main/java/org/eclipse/californium/scandium/DTLSConnector.java]: Ok except, that sounds good to merge.

Comment (boaks) [reply scandium-core/src/main/java/org/eclipse/californium/scandium/DTLSConnector.java]: so remove the "if"?

Comment (sbernard31) [reply scandium-core/src/main/java/org/eclipse/californium/scandium/DTLSConnector.java]: yes.

**Pull request merged**